Source/Game/Engine/Physics/ColliderData.cpp

```diff
@@ -9,25 +9,38 @@ AABBData::AABBData(Vector2 lowerBound, const Vector2 upperBound) : lowerBound(st
 
 AABBData::AABBData(Vector2 size) : AABBData(-Fixed::OneHalf * size, Fixed::OneHalf * size) {}
 
-bool AABBData::Intersecting(const AABBData&) const
+bool AABBData::Intersecting(const AABBData& other) const
 {
-	return false;
+	bool xInRange = (other.lowerBound.x >= this->lowerBound.x && other.lowerBound.x <= this->upperBound.x)
+					|| (other.upperBound.x >= this->lowerBound.x && other.upperBound.x <= this->upperBound.x);
+	bool yInRange = (other.lowerBound.y >= this->lowerBound.y && other.lowerBound.y <= this->upperBound.y)
+					|| (other.upperBound.y >= this->lowerBound.y && other.upperBound.y <= this->upperBound.y);
+	return xInRange && yInRange;
 }
 
-bool AABBData::IsPointInRect(Vector2) const
+bool AABBData::IsPointInRect(Vector2 point) const
 {
-	return false;
+	bool xInRange = (point.x >= this->lowerBound.x && point.x <= this->upperBound.x);
+	bool yInRange = (point.y >= this->lowerBound.y && point.y <= this->upperBound.y);
+	return xInRange && yInRange;
 }
 
 CircleData::CircleData(Fixed radius, Vector2 centre) : centre(std::move(centre)), radius(std::move(radius)) {}
 
-bool CircleData::IsIntersecting(const CircleData&) const
+bool CircleData::IsIntersecting(const CircleData& other) const
 {
-	return false;
+	auto length = other.radius + this->radius;
+	auto x = other.centre.x - this->centre.x;
+	auto y = other.centre.y - this->centre.y;
+
+	return (length * length) > (x * x) + (y * y);
 }
 
-bool CircleData::IsPointInCircle(Vector2) const
+bool CircleData::IsPointInCircle(Vector2 point) const
 {
-	return false;
+	auto x = point.x - this->centre.x;
+	auto y = point.y - this->centre.y;
+
+	return (this->radius * this->radius) > (x * x) + (y * y);
```

Comment:
Unfortunately `Fixed` will overflow here, being squared and all; so you'll have to use `ToF64()` / `ToF32()` before squaring each quantity, and compare native floats/doubles that way.

Try this:
```
f64 centreDist = (other.centre - centre).SqrMagnitude();
f64 radiiDist = (other.radius + radius).ToF64();
```

Reply:
And I can just subtract vectors like that? I saw Vector2 had some operator definitions I think, but I wasn't sure how they exactly worked.

Reply:
Yeah, add, subtract, Dot, are all supported, and I'd recommend not splitting up a Vector2 until necessary - it's a lot easier to think in terms of them too, esp when with orientations!